connectivity/check/policy.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020-2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/scheme"
+
+	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CiliumNetworkPolicy implements ConnectivityTest interface so that
+// policies can be applied between tests and policy apply failures can
+// be reported like any other test results.
+type CiliumNetworkPolicy struct {
+	Title  string
+	Policy string
+}
+
+// Name returns the absolute name of the policy
+func (p *CiliumNetworkPolicy) Name() string {
+	return p.Title
+}
+
+// Run applies the policy, use nil policy to delete all policies
+func (p *CiliumNetworkPolicy) Run(ctx context.Context, c TestContext) {
+	failures := c.ApplyPolicyYaml(ctx, p.Policy)
+	c.Report(TestResult{
+		Name:     p.Name(),
+		Failures: failures,
+		Warnings: 0,
+	})
+}
+
+type WithPolicy struct {
+	err  error
+	cnps []*ciliumv2.CiliumNetworkPolicy
+}
+
+func (wp *WithPolicy) Parse(policy string) {
+	wp.cnps, wp.err = ParsePolicyYaml(policy)
+}
+
+// Apply returns the number of failures and a cancel function that deletes the applied policies
+func (wp *WithPolicy) Apply(ctx context.Context, c TestContext) (failures int, cancel func()) {
+	if wp.err != nil {
+		return 1, func() {}
+	}
+
+	for _, cnp := range wp.cnps {
+		failures += c.ApplyCNP(ctx, cnp)
+	}
+	return failures, func() {
+		for _, cnp := range wp.cnps {
+			c.DeleteCNP(ctx, cnp)
+		}
+	}
+}
+
+// Run applies the policy, use empty policy to delete all policies
+func (k *K8sConnectivityCheck) ApplyPolicyYaml(ctx context.Context, policy string) int {
```

Comment:
Nit: use all caps for acronyms. Also, the func name in the godoc comments is wrong.

```suggestion
// ApplyPolicyYAML applies the policy, use empty policy to delete all policies
func (k *K8sConnectivityCheck) ApplyPolicyYAML(ctx context.Context, policy string) int {
```

Reply:
Thanks!